extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -363,30 +366,41 @@ public void run()
           // The retrying business is because the KafkaConsumer throws OffsetOutOfRangeException if the seeked-to
           // offset is not present in the topic-partition. This can happen if we're asking a task to read from data
           // that has not been written yet (which is totally legitimate). So let's wait for it to show up.
-          final ConsumerRecords<byte[], byte[]> records = RetryUtils.retry(
-              new Callable<ConsumerRecords<byte[], byte[]>>()
-              {
-                @Override
-                public ConsumerRecords<byte[], byte[]> call() throws Exception
+          ConsumerRecords<byte[], byte[]> records = ConsumerRecords.empty();
+          try {
+            polling = true;
```

Comment:
hmm, I think could this be problematic in the case where:

- [run thread] possiblyPause runs and doesn't pause
- [http thread] /pause is called and sets pauseRequested; it doesn't set polling = false or interrupt because polling is already false
- [run thread] read loop starts and never gets interrupted

I think we can avoid this by:

- getting rid of pollingSyncLock and instead synchronizing `polling` under the existing `pauseLock`
- when acquiring the lock for update polling = true in the main thread, do another possiblyPause before actually polling.

Or, maybe cleaner overall would be to stop using RetryUtils.retry, and instead implement retries through timed `await` on a condition. Then we also won't need to mess about with interrupts at all, we can just set a flag and signal the condition to interrupt a potential retry.